src/current/v23.1/automatic-cpu-profiler.md

```diff
@@ -0,0 +1,77 @@
+- Set `server.cpu_profile.interval` to a lower value, for example `1m40s` (1 minute 40 seconds).
+
+### High-water mark threshold 
+
+The Automatic CPU Profiler runs asynchronously in the background. After every second, the Automatic CPU Profiler checks if the CPU usage exceeds the high-water mark threshold. If so, it captures a CPU profile. If a profile capture is already in progress, a second profile is not taken.
+
+The Automatic CPU Profiler uses the configuration options to determine the high-water mark threshold. For example, with `duration` set to `10s` , `interval` set to `5m0s`, and `cpu_usage_combined_threshold` set to `80`:
+
+- At `time0` the CPU usage polled is 82 percent which exceeds the baseline threshold of `80`, so a `10s` profile is captured and the high-water mark threshold becomes `82`.
+- After the `10s` profile capture, the Automatic CPU Profiler continues to check every second if the CPU usage now exceeds `82` percent.
+- At `time1` the CPU usage polled is 85 percent, another profile is taken for `10s` and the high-water mark threshold becomes `85`.
+- At `time2`, the `5m0s` interval after `time0`, the high-water mark threshold is reset to the baseline threshold of `80`.
+- The Automatic CPU Profiler continues its every second polling and captures profiles when CPU usage exceeds the high-water mark threshold beginning at the `80` percent baseline.
+
+## Accessing CPU profiles
+
+- The Automatic CPU Profiler saves the captured CPU profiles to disk in the node filesystem. The default path is `/cockroach-data/logs/pprof_dump`.
+- The only way to get these CPU profiles is to access the node filesystem.
+- Enabling the Automatic CPU Profiler does **not** add CPU profiles to [debug zips](cockroach-debug-zip.html).
+
+## Overhead
+
+Enabling the automatic CPU profile capture on a cluster will add overhead to the cluster in the form of potential increases in latency and CPU usage. The decision to enable this feature should be done when advised by the [Cockroach Labs support team](support-resources.html).
+
+- Monitor the following metrics:
+  - [P99 latency](ui-sql-dashboard.html#service-latency-sql-99th-percentile)
+  - P50 latency by creating a [custom chart](ui-custom-chart-debug-page.html) for the `sql.exec.latency-p50` metric
+  - [CPU usage](ui-hardware-dashboard.html#cpu-percent)
+- We anticipate a sub-10% regression on these foreground latency metrics. This overhead to your cluster may be deemed acceptable in order to collect CPU profiles that are necessary to troubleshoot problems in your cluster. Please consult with Cockroach Labs Support.
+- Overhead only occurs during profile capture, not when it is idle. The default of `server.cpu_profile.duration` is 1 second. If the cluster can tolerate a `server.cpu_profile.duration` increase in latency to capture the CPU profile, consider enabling the Automatic CPU Profiler.
```

Comment:
> The default of `server.cpu_profile.duration` is 1 second. 

Nit: The default isn't 1 second. Suggest we cut out this sentence.

Reply:
Sorry about that incorrect rewrite. I have changed it to: "If the cluster can tolerate a server.cpu_profile.duration (for example, 1 second) increase in latency to capture the CPU profile, consider enabling the Automatic CPU Profiler."